support-figma/extended-layout-plugin/code.ts

```diff
@@ -741,10 +741,19 @@ else if (figma.command === "move-plugin-data") {
     rotation = (node as FrameNode).rotation;
 
     if (node.type == "FRAME" || node.type == "RECTANGLE") {
-      if (meterData && meterData.progressBarData && meterData.progressBarData.vertical)
-        progress = node.height;
-      else
-        progress = node.width;
+      if (meterData && meterData.progressBarData) {
```

Comment:
I wonder if there's some destructuring sugar like `if let Some(ProgressBarData::Vertical { end } = ...` in TypeScript so these can be less nesty. (Maybe ["Default Values" in this guide?](https://medium.com/@satyamsaluja04/destructuring-in-javascript-aliases-nested-objects-and-default-values-8fc1b378cf6d)). Maybe that makes it less readable?

Reply:
Yea I can make this a little better, will clean up.